plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -0,0 +1,760 @@
+                                 .build();
+
+        Deployment deployment = new DeploymentBuilder()
+                .withNewMetadata()
+                    .withName(dName)
+                    .withNamespace(this.cheOpenShiftProjectName)
+                .endMetadata()
+                .withNewSpec()
+                    .withReplicas(1)
+                    //.withTrigger(DeploymentTriggerType.CONFIG_CHANGE) ?
+                    .withNewSelector()
+                        .withMatchLabels(selector)
+                    .endSelector()
+                    .withNewTemplate()
+                        .withNewMetadata()
+                            .withLabels(selector)
+                        .endMetadata()
+                        .withSpec(podSpec)
+                    .endTemplate()
+                .endSpec()
+                .build();
+
+        deployment = openShiftClient.extensions()
+                                    .deployments()
+                                    .inNamespace(this.cheOpenShiftProjectName)
+                                    .create(deployment);
+
+        LOG.info(String.format("OpenShift deployment %s created", dName));
+        return deployment.getMetadata().getName();
+    }
+
+    private List<VolumeMount> getVolumeMountsFrom(String[] volumes, String workspaceID) {
+        List<VolumeMount> vms = new ArrayList<>();
+        for (String volume : volumes) {
+            String mountPath = volume.split(":",3)[1];
+            String volumeName = getVolumeName(volume);
+
+            VolumeMount vm = new VolumeMountBuilder()
+                    .withMountPath(mountPath)
+                    .withName("ws-" + workspaceID + "-" + volumeName)
+                    .build();
+            vms.add(vm);
+        }
+        return vms;
+    }
+
+    private List<Volume> getVolumesFrom(String[] volumes, String workspaceID) {
+        List<Volume> vs = new ArrayList<>();
+        for (String volume : volumes) {
+            String hostPath = volume.split(":",3)[0];
+            String volumeName = getVolumeName(volume);
+
+            Volume v = new VolumeBuilder()
+                    .withNewHostPath(hostPath)
+                    .withName("ws-" + workspaceID + "-" + volumeName)
+                    .build();
+            vs.add(v);
+        }
+        return vs;
+    }
+
+    private String getVolumeName(String volume) {
+        if (volume.contains("ws-agent")) {
+            return "wsagent-lib";
+        }
+
+        if (volume.contains("terminal")) {
+            return "terminal";
+        }
+
+        if (volume.contains("workspaces")) {
+            return "project";
+        }
+
+        return "unknown-volume";
+    }
+
+    private String waitAndRetrieveContainerID(String deploymentName) {
+        for (int i = 0; i < OPENSHIFT_WAIT_POD_TIMEOUT; i++) {
+            try {
+                Thread.sleep(OPENSHIFT_WAIT_POD_DELAY);
```

Comment:
Well, we do need to wait till the pod status will be "Running" here. Do you mean should we decrease the "OPENSHIFT_WAIT_POD_DELAY" time (1 second for now) ?